recipes/perl/access-control/create_access_control_role.pl

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env perl
+
+use LWP::UserAgent;
+use LWP::Protocol::https;
+use JSON;
+use Getopt::Long;
+
+require 'api_requests.pl';
+require 'access_control_api_requests.pl';
+
+my $hostname;
+my $username;
+my $password;
+my $domain_name;
+my $domain_type;
+my $role_name;
+my $role_description;
+
+my $base_url;
+my $token;
+
+GetOptions(
+    'hostname=s' => \$hostname,
+    'username=s' => \$username,
+    'password=s' => \$password,
+    'domain_name=s' => \$domain_name,
+    'domain_type=s' => \$domain_type,
+    'role_name=s' => \$role_name,
+    'role_description=s' => \$role_description
+) or die print_usage();
```

Comment:
It's more common to to use `pod2usage` instead of a bespoke usage statement. That also avoids the peculiarity of using `die`, which expects to receive an error message as its argument, on the result of a function that doesn't return a message.

```suggestion
) or pod2usage(2);
```

Reply:
Updated